### PR TITLE
Revert the rest of #69591

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
None

#### Purpose of change
Removes calorie gain from photosynthesis as it was added based on flawed premises.
PR adding it uses the concept of "leaf area index" to calculate the total leaf area of the mutant, but then inappropriately drops that area into a calculation based on solar insolation, which is not a comparable value. There are some other questionable arguments as well, resulting in the mind-boggling value of 10 square meters of captured sunlight.

Removes a bunch of photosynthesis changes entangled with the calorie gain.
I can't see how to extricate the calorie gain changes from the pile of other features that impact photosynthesis without starting from scratch, and I'm not doing that, so it's all out.

Removes the seasonal variation in leaves because it's already a huge amount of content for some very minor impacts and this is not a maintainable pattern.
This pattern of establishing multiple variants of the same mutation based on season and swapping between them with EOCs is ok for singular mutations with two variants, but applying it to a whole series of mutations (leaves1-2-3 and vine arms1-2-3 and rosebuds, and creating variants across each season creates an unmaintainable pile of interrelated traits and makes a mess in the code where you have many synonyms for each trait you have to list when testing for them.

Completes removal of fruiting and related features. It's broken and needs heavy rework if it will go in.
The tree growth code was broken and was already reverted, rendering most of this dead code/content, this removes the rest.

#### Describe the solution
Revert

### CODE REVIEW

**Comment by Maleclypse** [comment]
bunch of missed transparencies in XE and MoM are failing your tests.

**Comment by anoobindisguise** [comment]
So if I'm understanding correctly, the original PR (inappropriately) bundled other seasonal features like S.A.D., seasonal allergies and fur shedding; the intention is to keep those but just remove photosynthesis correct?

**Comment by kevingranade** [comment]
Yes I'm intentionally leaving SAD and hay fever and shedding in place and just excising the specifically problematic bits.

**Comment by NancokPS2** [comment]
So the calorie gain is removed right?  
Because the description still says they let you photosynthesize.  
Unless it still has some other effect (?)

**Comment by I-am-Erk** [comment]
> So the calorie gain is removed right? Because the description still says they let you photosynthesize. Unless it still has some other effect (?)

I'd have to check what Kevin's up to here but I'd assume he's reverting it to where it was before, ie. still causes calorie gain but not as much. Correct me if I'm wrong, it's been a good long time since I looked at photosynthesis.

**Comment by I-am-Erk** [comment]
> So if I'm understanding correctly, the original PR (inappropriately) bundled other seasonal features like S.A.D., seasonal allergies and fur shedding; the intention is to keep those but just remove photosynthesis correct?

For onlookers confused about what's up here, I really want to emphasize that this is a lot of the problem. When I see a PR title about adding seasonal mutations, I do not expect to be looking for photosynthesis calorie changes, conversations with fungal trees, extensive rebalancing of clothing transparency, etc.

This is not the original PR author's fault, as I stated in an earlier revert. This PR shouldn't have been merged as is, and that's on me and kevin for not adequately communicating with our team. Many of these changes might be re-evaluated and added back in in some form, but almost all of them should have been individual small PRs allowing us to examine and discuss the changes. Once we've gone through and spotted a number of problems with the PR on a core level it's basically just time to revert it and rework it from the start before it gets entrenched. We're working on ways to avoid this in the future.

**Comment by kevingranade** [comment]
>  Because the description still says they let you photosynthesize.

I should have been clearer with the summary, this rolls back everything to do with photosynthesis to where it was previously, there are some circumstances where you can gain *some* calories, and other circumstances where you're getting meaningful amounts of vitamins out of the process, but this rolls back the very significant increase that was added in the Seasonal EOCs PR.

**Comment by carlarctg** [comment]
> > So if I'm understanding correctly, the original PR (inappropriately) bundled other seasonal features like S.A.D., seasonal allergies and fur shedding; the intention is to keep those but just remove photosynthesis correct?
> 
> For onlookers confused about what's up here, I really want to emphasize that this is a lot of the problem. When I see a PR title about adding seasonal mutations, I do not expect to be looking for photosynthesis calorie changes, conversations with fungal trees, extensive rebalancing of clothing transparency, etc.
> 
> This is not the original PR author's fault, as I stated in an earlier revert. This PR shouldn't have been merged as is, and that's on me and kevin for not adequately communicating with our team. Many of these changes might be re-evaluated and added back in in some form, but almost all of them should have been individual small PRs allowing us to examine and discuss the changes. Once we've gone through and spotted a number of problems with the PR on a core level it's basically just time to revert it and rework it from the start before it gets entrenched. We're working on ways to avoid this in the future.

If there's any interest on the dev team's side in softening the blow, these reverts could be _applied over time_ rather than all at once, possibly with the rework bundled, because as-is, throwing all the reverts at the wall, feels much, much more directed and aggressive a path to take. Right now, it's very demotivating to see all these reverts so quickly after all these discussions that seem to have ended up in kind words from you and little from anywhere else. The 'optimal' path that excises the issues within the codebase as quickly as possible also ends up worsening the growing issues within the contributor community. 

The end result is not just losing the motivated contributor's future contributions but also a cascade of further contribution loss by others that hear of the situation and feel its unjust, losing the drive to help improve the game. Losing hundreds of potential PRs, small and large, that help shape the game towards the desired ideal, all just because of unnecessary haste and poor communication, is basically a tragedy.

I don't mean this motivation loss towards fairy/wormy - that's a given. I mean the contributor community. I can attest for myself that I lost motivation in contributing and have seen at least two highly active contributors share that sentiment openly. Imagine how many just don't speak up and yet feel the same?

> We're working on ways to avoid this in the future.

Perhaps there's discussion and changes in the background, inside the developmental team that we just can't see, Talking about how to make sure things like this stop happening. That would be nice, but in terms of PR, amounts to almost nothing if the exterior looks the same from one day to another. The fears of being 'stonewalled' still exist. Without visible change, be it an announcement, or an otherwise highly public change in how developers do things, these 'ways' will prevent the next crisis, but will do nothing to help with the aftermath of the current one.

**Comment by I-am-Erk** [comment]
There's an active motivation to revert any major problems as quickly as possible because if we don't, people are liable to start making changes that are dependent on this material, and we'll have to revert more. The original PR author here has a tendency to huge omnibus PRs as I just described, and those are the kind of PRs where if something sneaks in, we're liable to just have to revert it and sort it out. Many devs and collabs have described this now in many PRs, I think at this point if that explanation is inadequate and you're continuing to feel like anything could get reverted at any time for any reason, there's probably no other way we can make it clear.

Most of the changes necessary for this are communication with the merge team, which obviously isn't likely to happen in a public space. However I've posted my in progress documentation changes to discord, you should see that PRd tonight, if I have time, or tomorrow morning if I don't.

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -207,7 +206,7 @@ static const json_character_flag json_flag_ASOCIAL1( "ASOCIAL1" );
 static const json_character_flag json_flag_ASOCIAL2( "ASOCIAL2" );
 static const json_character_flag json_flag_CANNIBAL( "CANNIBAL" );
 static const json_character_flag json_flag_PAIN_IMMUNE( "PAIN_IMMUNE" );
-static const json_character_flag json_flag_PSYCHOPATH( "PSYCHOPATH" );
+static const json_character_flag json_flag_PSYCHOPATH( "PAIN_IMMUNE" );
```
Comment (Fris0uman) [inline src/activity_handlers.cpp:209]:
```suggestion
static const json_character_flag json_flag_PSYCHOPATH( "PSYCHOPATH" );
```

Comment (kevingranade) [reply src/activity_handlers.cpp]: How the heck did I manage to do that.

**Comment by kevingranade** [comment]
Concrete example of why, "trickle the reversions out over time" isn't workable.

([slow] ~starting_items)=> (continued from above) ERROR : src/item_factory.cpp:729 [auto Item_factory::finalize_post(itype &)::(anonymous class)::operator()(const flag_id &) const] itype 'integrated_feathers' uses undefined flag 'TRANSPARENT'. Please add corresponding 'json_flag' entry to json.

**Pull request merged**